content/configuration/communication/vault-csi.md

```diff
@@ -0,0 +1,139 @@
+    helm repo add hashicorp https://helm.releases.hashicorp.com
+    helm repo update
+    helm install vault hashicorp/vault --namespace default \
+        --set "server.dev.enabled=true" \
+        --set "injector.enabled=false" \
+        --set "csi.enabled=true"
+    ```
+
+2. Add Slack token in Vault:
+    ```bash
+    # Exec to pod
+    kubectl exec -n default -it vault-0 -- /bin/sh
+    ```
+    ```bash
+    # Write the token to Vault
+    vault kv put secret/slack token={token}
+    ```
+
+3. Enable Vault's Kubernetes authentication:
+    ```bash
+    vault auth enable kubernetes
+    vault write auth/kubernetes/config \
+        kubernetes_host="https://$KUBERNETES_PORT_443_TCP_ADDR:443"
+    ```
+    ```bash
+    vault policy write internal-app - <<EOF
+    path "secret/data/slack" {
+      capabilities = ["read"]
+    }
+    EOF
+    ```
+    ```bash
+    vault write auth/kubernetes/role/database \
+        bound_service_account_names=botkube-sa \
+        bound_service_account_namespaces=default \
+        policies=internal-app \
+        ttl=20m
+    ```
+
+4. Install the Secrets Store CSI driver:
+    ```bash
+    helm repo add secrets-store-csi-driver https://kubernetes-sigs.github.io/secrets-store-csi-driver/charts
+    helm install csi secrets-store-csi-driver/secrets-store-csi-driver \
+    --namespace default \
+    --set syncSecret.enabled=true
+    ```
+
+5. Exit from the Vault Pod:
+    ```bash
+    exit
+    ```
+
+6. Create BotKue installation parameters:
+    ```yaml
+    cat > /tmp/values.yaml << ENDOFFILE
+    extraObjects:
+      - apiVersion: secrets-store.csi.x-k8s.io/v1
+        kind: SecretProviderClass
+        metadata:
+          name: vault-database
+        spec:
+          provider: vault
+          secretObjects:
+            - data:
+                - key: token
+                  objectName: "slack-token"
+              secretName: communication-slack
+              type: Opaque
+          parameters:
+            vaultAddress: "http://vault.default:8200"
+            roleName: "database"
+            objects: |
+              - objectName: "slack-token"
+                secretPath: "secret/data/slack"
+                secretKey: "token"
+
+    communications:
+      # Settings for Slack
+      slack:
+        enabled: true
+        channel: 'random'
+        # token - specified via env variable
+
+    extraEnv:
+      - name: COMMUNICATION_SLACK_TOKEN
+        valueFrom:
+          secretKeyRef:
+            name: communication-slack
+            key: token
+
+    extraVolumeMounts:
+      - name: secrets-store-inline
+        mountPath: "/mnt/secrets-store"
+        readOnly: true
+
+    extraVolumes:
+      - name: secrets-store-inline
+        csi:
+          driver: secrets-store.csi.k8s.io
+          readOnly: true
+          volumeAttributes:
+            secretProviderClass: "vault-database"
+    ENDOFFILE
+    ```
+
+7. Install BotKube:
+		<!--- replace the version with v0.13.0 once released -->
```

Comment:
It would be good to add some kind of prefix to this comment (e.g. `TODO(https://github.com/infracloudio/botkube/issues/595):`).

BTW, please update the task to make sure we won't forget about it 🙂 
https://github.com/infracloudio/botkube/issues/595